corehq/apps/analytics/utils.py

```diff
@@ -195,34 +193,68 @@ def _get_contact_ids_to_delete(list_of_emails, retry_num=0):
                 return []
             req.raise_for_status()
         except (ConnectionError, requests.exceptions.HTTPError) as e:
-            metrics_counter(
-                'commcare.hubspot_data.retry.get_contact_ids_for_emails'
-            )
+            if record_metrics:
+                metrics_counter(
+                    'commcare.hubspot_data.retry.get_contact_ids_for_emails'
+                )
             if retry_num <= MAX_API_RETRIES:
-                return _get_contact_ids_to_delete(list_of_emails, retry_num + 1)
-            else:
+                return _get_contacts_from_hubspot(list_of_emails, retry_num + 1)
+            elif record_metrics:
                 metrics_counter(
                     'commcare.hubspot_data.error.get_contact_ids_for_emails',
                     tags={
                         'error': str(e),
                     }
                 )
         else:
-            ids_to_delete = []
-            for contact_id, data in req.json().items():
-                first_conversion_status = data.get(
-                    'properties', {}
-                ).get('first_conversion_clustered_', {}).get('value')
-
-                # If a user's first conversion IS in the allowed list, it means
-                # they have directly interacted with us and we want to keep them
-                # in the email list. However, we will not send project data
-                # about them to HubSpot as they will still be blocked from
-                # updates in track_periodic_data.
-                if first_conversion_status not in ALLOWED_CONVERSIONS:
-                    ids_to_delete.append(contact_id)
-            return ids_to_delete
-    return []
+            return req.json()
+    return {}
+
+
+def get_first_conversion_status_for_emails(list_of_emails):
+    """
+    This returns the First Conversion (clustered) property for each contact in
+    list_of_emails if that email is present in HubSpot
+    :param list_of_emails:
+    :return: list of contact ids
+    """
+    hubspot_contacts = _get_contacts_from_hubspot(
+        list_of_emails, record_metrics=False
+    )
+    status_summary = {}
+    for contact_id, data in hubspot_contacts.items():
+        first_conversion_status = data.get(
+            'properties', {}
+        ).get('first_conversion_clustered_', {}).get('value')
+        email = data.get(
+            'properties', {}
+        ).get('email', {}).get('value')
+        status_summary[email] = first_conversion_status
+    return status_summary
+
+
+def _get_contact_ids_to_delete(list_of_emails):
```

Comment:
Continuing with the spirit of the newspaper metaphor, this could be moved below the functions that call it (which are below).